src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,337 @@
+pub struct Frp {
+    pub inputs  : FrpInputs,
+    pub outputs : FrpOutputs,
+    pub network : frp::Network,
+}
+
+impl Deref for Frp {
+    type Target = FrpInputs;
+    fn deref(&self) -> &Self::Target {
+        &self.inputs
+    }
+}
+
+impl Default for Frp {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Frp {
+    /// Create new Frp.
+    pub fn new() -> Self {
+        let network = frp::Network::new();
+        let inputs  = FrpInputs::new(&network);
+        let outputs = FrpOutputs::new(&network);
+        Self{network,inputs,outputs}
+    }
+}
+
+
+
+// ==================
+// === Animations ===
+// ==================
+
+/// Breadcrumbs' animations handlers.
+#[derive(Debug,Clone,CloneRef,Copy)]
+pub struct Animations {}
+
+impl Animations {
+    /// Create new animations handlers.
+    pub fn new(_network:&frp::Network) -> Self {
+        Self{}
+    }
+}
+
+
+
+// ========================
+// === BreadcrumbsModel ===
+// ========================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct BreadcrumbsModel {
+    logger                : Logger,
+    animations            : Animations,
+    display_object        : display::object::Instance,
+    pub project_name      : ProjectName,
+    breadcrumbs_container : display::object::Instance,
+    scene                 : Scene,
+    breadcrumbs           : Rc<RefCell<Vec<Breadcrumb>>>,
+    breadcrumb_pop        : frp::Source,
+    breadcrumb_push       : frp::Source<(Rc<MethodPointer>,uuid::Uuid)>,
+    current_index         : Rc<Cell<usize>>
+}
+
+impl BreadcrumbsModel {
+    /// Create new BreadcrumbsModel.
+    pub fn new<'t,S:Into<&'t Scene>>(scene:S, frp:&Frp, focus_manager:&FocusManager) -> Self {
+        let scene                 = scene.into();
+        let project_name          = ProjectName::new(scene,focus_manager);
+        let logger                = Logger::new("Breadcrumbs");
+        let display_object        = display::object::Instance::new(&logger);
+        let breadcrumbs_container = display::object::Instance::new(&logger);
+        let animations            = Animations::new(&frp.network);
+        let scene                 = scene.clone_ref();
+        let breadcrumbs           = default();
+        let breadcrumb_pop        = frp.outputs.breadcrumb_pop.clone_ref();
+        let breadcrumb_push       = frp.outputs.breadcrumb_push.clone_ref();
+        let current_index         = default();
+        Self{logger,display_object,animations,scene,breadcrumbs,breadcrumb_pop,breadcrumb_push,
+            project_name,breadcrumbs_container,current_index}.init()
+    }
+
+    fn init(self) -> Self {
+        self.add_child(&self.project_name);
+        self.add_child(&self.breadcrumbs_container);
+        self.project_name.set_position(Vector3::new(HORIZONTAL_MARGIN,0.0,0.0));
+        self.set_project_name_width(self.project_name.width());
+        //FIXME[dg]: This is a workaround. Main shouldn't be a MethodPointer.
```

Comment:
Please write a much better explaanation here. I do not understand it. All FIXMEs should be understandable even by people who do not know other parts of the codebase.